src/docs/development/platform-integration/platform-channels.md

```diff
@@ -637,6 +636,52 @@ If using the iOS Simulator,
 note that it does not support battery APIs,
 and the app displays 'battery info unavailable'.
 
+## Typesafe platform channels via Pigeon {#pigeon}
+
+The previous example uses `MethodChannel` to communicate between the host and
+client which isn't typesafe.  Calling and receiving messages depends on the host
+and client declaring the same arguments and datatypes in order for messages to
+work.  The [Pigeon][] package can be used as an alternative to `MethodChannel`
+to generate code that sends messages in a typesafe manner.
```

Comment:
Also mention you can send non-primitives

Reply:
s/typesafe messages/structured typesafe messages